extensions/Worker.Extensions.Shared/Exceptions/InvalidBindingSourceException.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions
+{
+    /// <summary>
+    /// The exception that is thrown when an invalid binding source is provided.
+    /// </summary>
+    public class InvalidBindingSourceException : InvalidOperationException
```

Comment:
Change this to `internal`, we can't put `public` in the shared project, as this will cause the same type to be available from every package which references this.

We should probably make an analyzer for that.

Reply:
Thanks for catching!